### PR TITLE
Use context managers to plug leaks of open file handles.

### PR DESCRIPTION
**Pull request by rodrigc**
Fixes this warning on Python 3:
    builtins.ResourceWarning: unclosed file

### CODE REVIEW

File: worker/buildbot_worker/test/util/misc.py [hunk 1]
```diff
@@ -112,13 +112,16 @@ def setUpOpen(self, file_contents="dummy-contents"):
...
                               read() method
         """
-        # create mocked file object that returns 'file_contents' on read()
-        # and tracks any write() calls
-        self.fileobj = mock.Mock()
-        self.fileobj.read = mock.Mock(return_value=file_contents)
-        self.fileobj.write = mock.Mock()
+        # Use mock.mock_open() to create a substitute for
+        # open().
+        fakeOpen = mock.mock_open(read_data=file_contents)
```
Comment (tardyp) [inline worker/buildbot_worker/test/util/misc.py]: good catch!

**Comment by tardyp** [comment]
@rutsky you can look here https://nine.buildbot.net/#/builders/2/builds/95465
the kind of false positive that are quite annoying.

**Comment by rodrigc** [comment]
I have changed the comment so pylint doesn't fail

**Comment by rutsky** [comment]
@tardyp @rodrigc looks like the only way to "fix" spelling in `# patch open() to always return self.fileobj` is to add `fileobj` to list of ignored words, I don't thing pylint is smart enough to distinguish code samples from comments/docstrings.
IMO it's OK to add such exceptions.

**Pull request merged**